packages/angular/cli/src/command-builder/command-module.ts

```diff
@@ -139,9 +139,17 @@ export abstract class CommandModule<T extends {} = {}> implements CommandModuleI
     }
 
     // Gather and report analytics.
+    let analyticsFlushPromise = Promise.resolve();
+    let analyticsFlushInterval: NodeJS.Timeout | undefined;
     const analytics = await this.getAnalytics();
+
     if (this.shouldReportAnalytics) {
       await this.reportAnalytics(camelCasedOptions);
+
+      // Flush on an interval (if the event loop is waiting).
+      analyticsFlushInterval = setInterval(() => {
+        analyticsFlushPromise = analyticsFlushPromise.then(() => analytics.flush());
```

Comment:
I assume this is the fix for watch-mode analytics. Would this be a better feature in the analytics module rather than in the command builder?

Reply:
This technically would need to be an API change as analytics API is in `@angular-devkit/core`.  https://github.com/angular/angular-cli/tree/main/packages/angular_devkit/core/src/analytics

I don't however see a completing reason why analytics API should expose a period flush method considering this is quite a specific case where a flush need to happen periodicly.